File: bench/frontend/src/types/product.ts

export interface ProductQuery {
  pageNo: number;
  pageSize: number;
  productNo?: string;
  productName?: string;
  categoryId?: number;
  status?: number;
}

export interface ProductVO {
  id: number;
  productNo: string;
  productName: string;
  categoryId: number;
  categoryName?: string;
  salePrice: number;
  stockQuantity: number;
  stockWarnThreshold: number;
  status: number;
  createdAt: string;
}

export interface ProductDetailVO {
  id: number;
  productNo: string;
  categoryId: number;
  categoryName?: string;
  productName: string;
  subTitle?: string;
  brandName?: string;
  unitName?: string;
  mainImage?: string;
  salePrice: number;
  costPrice: number;
  marketPrice: number;
  stockQuantity: number;
  stockWarnThreshold: number;
  status: number;
  auditStatus: number;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface ProductForm {
  productNo?: string;
  categoryId: number;
  productName: string;
  subTitle?: string;
  brandName?: string;
  unitName?: string;
  mainImage?: string;
  salePrice: number;
  costPrice: number;
  marketPrice: number;
  stockQuantity: number;
  stockWarnThreshold: number;
  remark?: string;
}

export interface ProductCategoryVO {
  id: number;
  parentId: number;
  categoryName: string;
  categoryCode: string;
  levelNo: number;
  sortNo: number;
  status: number;
}

File: bench/src/main/java/com/ec/workbench/module/product/entity/Product.java
package com.ec.workbench.module.product.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Product {
    private Long id;
    private Long merchantId;
    private String productNo;
    private Long categoryId;
    private String productName;
    private String subTitle;
    private String brandName;
    private String unitName;
    private String mainImage;
    private BigDecimal salePrice;
    private BigDecimal costPrice;
    private BigDecimal marketPrice;
    private Integer stockQuantity;
    private Integer stockLocked;
    private Integer stockWarnThreshold;
    private Integer saleCount;
    private Integer status;
    private Integer auditStatus;
    private Integer isDeleted;
    private String remark;
    private LocalDateTime createdAt;
    private Long createdBy;
    private LocalDateTime updatedAt;
    private Long updatedBy;
}
